tests/unit/utils/test_urlutils.py

```diff
@@ -416,6 +424,18 @@ def test_is_url(config_stub, fake_dns,
             auto_search))
 
 
+@pytest.mark.parametrize('auto_search, open_base_url, is_url', [
+    ('naive', True, False),
+    ('naive', False, False),
+    ('never', True, False),
```

Comment:
I fixed that kind of accidentially so I added a test. There's still a dns lookup with `url.auto_search=dns` though, but maybe that's enough for now.